hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssMessageHandler.java

```diff
@@ -56,6 +71,44 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
         requireNonNull(context);
-        submissionManager.submitTssVote(TssVoteTransactionBody.DEFAULT, context);
+        final var op = context.body().tssMessageOrThrow();
+
+        final var tssStore = context.storeFactory().writableStore(WritableTssBaseStore.class);
+        final var rosterStore = context.storeFactory().readableStore(ReadableRosterStore.class);
+        final var maxSharesPerNode =
+                context.configuration().getConfigData(TssConfig.class).maxSharesPerNode();
+        final var numberOfAlreadyExistingMessages =
+                tssStore.getTssMessages(op.targetRosterHash()).size();
+
+        // The sequence number starts from 0 and increments by 1 for each new message.
+        final var key = TssMessageMapKey.newBuilder()
+                .rosterHash(op.targetRosterHash())
+                .sequenceNumber(numberOfAlreadyExistingMessages)
+                .build();
+        // Each tss message is stored in the tss message state and is sent to CryptographyManager for further
+        // processing.
+        tssStore.put(key, op);
+
+        final var tssParticipantDirectory =
+                computeTssParticipantDirectory(rosterStore.getActiveRoster(), maxSharesPerNode, (int)
+                        context.networkInfo().selfNodeInfo().nodeId());
+        final var result = tssCryptographyManager.handleTssMessageTransaction(op, tssParticipantDirectory, context);
+
+        result.thenAccept(ledgerIdAndSignature -> {
```

Comment:
I think the invocation of thenAccept could cause a NPE in an edge case. `TssCryptographyManager::computePublicKeysAndSignIfThresholdMet` could return null, then `handleTssMessageTransaction` will return a CompletableFuture that completes with null. Then in `TssMessageHandler`, the result.thenAccept would cause an NPE?

Reply:
I think if we return `CompletedFuture` with `null` , `thenAccept` wont cause NPE. And Since there is check for `ledgerIdAndSignature != null` I think this should not be any issue.